warehouse/query-core/src/main/java/datawave/query/tables/keyword/KeywordQueryLogic.java

```diff
@@ -0,0 +1,413 @@
+ */
+public class KeywordQueryLogic extends BaseQueryLogic<Entry<Key,Value>> implements CheckpointableQueryLogic {
+
+    private static final Logger log = Logger.getLogger(KeywordQueryLogic.class);
+
+    /**
+     * Used to pull back specific views of the data - expected to be a comma-delimited list of one or more views, if present.
+     */
+    public static final String CONTENT_VIEW_NAMES = "content.view.names";
+
+    /**
+     * Used to embed document language in query term
+     */
+    public static final String LANGUAGE_TOKEN = "%LANGUAGE:";
+
+    private static final String PARENT_ONLY = "\1";
+    private static final String ALL = "\u10FFFF";
+
+    private int queryThreads = 100;
+
+    ScannerFactory scannerFactory;
+
+    private KeywordQueryConfiguration config;
+
+    public KeywordQueryLogic() {
+        super();
+    }
+
+    public KeywordQueryLogic(final KeywordQueryLogic keywordQueryLogic) {
+        super(keywordQueryLogic);
+        this.queryThreads = keywordQueryLogic.queryThreads;
+        this.scannerFactory = keywordQueryLogic.scannerFactory;
+        this.config = new KeywordQueryConfiguration(keywordQueryLogic.config);
+    }
+
+    /**
+     * This method calls the base logic's close method, and then attempts to close all batch scanners tracked by the scanner factory, if it is not null.
+     */
+    @Override
+    public void close() {
+        super.close();
+        final ScannerFactory factory = this.scannerFactory;
+        if (null == factory) {
+            log.debug("ScannerFactory is null; not closing it.");
+        } else {
+            int nClosed = 0;
+            factory.lockdown();
+            for (final ScannerBase bs : factory.currentScanners()) {
+                factory.close(bs);
+                ++nClosed;
+            }
+            if (log.isDebugEnabled())
+                log.debug("Cleaned up " + nClosed + " batch scanners associated with this query logic.");
+        }
+    }
+
+    @Override
+    public GenericQueryConfiguration initialize(final AccumuloClient client, final Query settings, final Set<Authorizations> auths) throws Exception {
+        // Initialize the config and scanner factory
+        // NOTE: This needs to set the class-level config object. Do not use a local instance!
+        config = new KeywordQueryConfiguration(this, settings);
+        config.setClient(client);
+        config.setAuthorizations(auths);
+        config.setState(new KeywordQueryState());
+
+        // Get the BYPASS_ACCUMULO parameter if given
+        String bypassAccumuloString = settings.findParameter(BYPASS_ACCUMULO).getParameterValue().trim();
+        if (StringUtils.isNotBlank(bypassAccumuloString)) {
+            boolean bypassAccumuloBool = Boolean.parseBoolean(bypassAccumuloString);
+            config.setBypassAccumulo(bypassAccumuloBool);
+        }
+
+        this.scannerFactory = new ScannerFactory(config);
```

Comment:
I led you wrong here. There is actually more work to be done to support this. The only code that currently supports the bypass flag is used via the PushdownScheduler: https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/query/scheduler/PushdownScheduler.java#L166

They key is instead of using a Scanner you need to use a ScannerSession. This allows you to set the setDelegatedInitializer() to RFileResource.class when bypassAccumulo is configured. Since you already have a ScannerFactory, you can get access to a ScannerSession but it will unfortunately require a bit more code.